products/cloudiot/terraform.yaml

```diff
@@ -145,7 +145,41 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       credentials.publicKeyCertificate: !ruby/object:Overrides::Terraform::PropertyOverride
         # See the comment on stateNotificationConfig.exclude
         exclude: true
-
+  Device: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: [ "{{%registry}}/devices/{{name}}" ]
+    properties:
+      id: !ruby/object:Overrides::Terraform::PropertyOverride
+        required: true
+        name: 'name'
+        description: |
+          A unique name for the resource.
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        # We don't need this field, because it has the same format as the ID
+        exclude: true
+      gatewayConfig: !ruby/object:Overrides::Terraform::PropertyOverride
+        # The only mutable gateway_config field is gateway_auth_method,
+        # at least according to the API responses.
+        update_mask_fields:
+          - "gateway_config.gateway_auth_method"
```

Comment:
Have you tested updating this field by building the provider and attempting to modify the gateway_config object?

Terraform tests will not catch issues around deleting/recreating resources vs updating them, so it may be passing by doing a full delete/recreate rather than actually updating the existing resource

Reply:
(which exercises that field btw)

Reply:
Looks like updating the auth method works, but this seems to allow updating the `gateway_type` without destroying and recreating the resource, which then produces a perma-diff. I think the rest of the fields on `gateway_config` besides this one that can be updated should be marked as `input`

Reply:
Done. The other non-output field is gatewayType. I marked it as input as requested.

The other two (`lastAccessedGatewayId` and `lastAccessedGatewayTime`) are output only.